java-checks/src/main/java/org/sonar/java/checks/TooManyMethodsCheck.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012 SonarSource
+ * sonarqube@googlegroups.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonar.java.checks;
+
+import com.google.common.collect.ImmutableList;
+import org.sonar.api.server.rule.RulesDefinition;
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+import org.sonar.check.RuleProperty;
+import org.sonar.plugins.java.api.tree.ClassTree;
+import org.sonar.plugins.java.api.tree.MethodTree;
+import org.sonar.plugins.java.api.tree.Tree;
+import org.sonar.squidbridge.annotations.ActivatedByDefault;
+import org.sonar.squidbridge.annotations.SqaleConstantRemediation;
+import org.sonar.squidbridge.annotations.SqaleSubCharacteristic;
+
+import java.util.List;
+
+
+@Rule(
+  key="S1448",
+        name = "Classes should not have too many methods",
+        tags = {"brain-overload"},
+        priority = Priority.MAJOR)
+@ActivatedByDefault
+@SqaleSubCharacteristic(RulesDefinition.SubCharacteristics.ARCHITECTURE_CHANGEABILITY)
+@SqaleConstantRemediation("1h")
+public class TooManyMethodsCheck extends SubscriptionBaseVisitor {
+
+  private static final int DEFAULT_MAXIMUM = 35;
+
+  @RuleProperty(
+    key = "maximumMethodThreshold",
+    description = "The maximum number of methods authorized in a class.",
+    defaultValue = "" + DEFAULT_MAXIMUM
+  )
+  public int maximumMethodThreshold=DEFAULT_MAXIMUM;
+
+  @RuleProperty(
+    key = "countNonpublicMethods",
+    description = "Whether or not to include non-public methods in the count.",
+    defaultValue = "true"
+  )
+  public boolean countNonPublic = true;
+
+  @Override
+  public List<Tree.Kind> nodesToVisit() {
+
+    return ImmutableList.of(Tree.Kind.CLASS, Tree.Kind.ENUM, Tree.Kind.INTERFACE);
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+
+    int count = 0;
+
+    ClassTree classTree = (ClassTree) tree;
+    List<Tree> members = classTree.members();
+    for (Tree member : members) {
+      if (member instanceof MethodTree) {
+        if (countNonPublic) {
+          count++;
+        } else if (((MethodTree) member).symbol().isPublic()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This branch's code block is the same as the block for the branch on line 78. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1871)